### PR TITLE
chore: archiver cleanup

### PR DESCRIPTION
**Pull request by fracasula**
# Description

Prepared statements for archiver + more DRY and cleanup.

## Linear Ticket

< [Linear Link](https://linear.app/rudderstack/issue/PIPE-121/warehouse-archiver-cleanup) >

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/archive/archiver.go [hunk 1]
```diff
@@ -96,7 +99,9 @@ func (a *Archiver) backupRecords(ctx context.Context, args backupRecordsArgs) (b
 		Config:   filemanagerutil.GetProviderConfigForBackupsFromEnv(ctx, config.Default),
 	})
 	if err != nil {
-		err = fmt.Errorf("error in creating a file manager for:%s. Error: %w", config.GetString("JOBS_BACKUP_STORAGE_PROVIDER", "S3"), err)
+		err = fmt.Errorf("error in creating a file manager for:%s. Error: %w",
```
Comment (lvrach) [inline warehouse/archive/archiver.go:102]: Better error message

File: warehouse/archive/archiver.go [hunk 2]
```diff
@@ -72,13 +72,16 @@ type Archiver struct {
 }
 
 func (a *Archiver) backupRecords(ctx context.Context, args backupRecordsArgs) (backupLocation string, err error) {
-	a.Logger.Infof(`Starting backupRecords for uploadId: %s, sourceId: %s, destinationId: %s, tableName: %s,`, args.uploadID, args.sourceID, args.destID, args.tableName)
+	a.Logger.Infof("[Archiver]: Starting backupRecords for uploadId: %s, sourceId: %s, destinationId: %s, tableName: %s,",
```
[the remark this answers is not in the document]
Comment (fracasula) [reply warehouse/archive/archiver.go:75]: I agree, I was merely following an already existing convention for this file. Let's address this properly once we have structured logging in place :+1:

File: warehouse/archive/archiver.go [hunk 3]
```diff
@@ -310,131 +273,176 @@ func (a *Archiver) Do(ctx context.Context) error {
...
+	loadLocationRows, err := txn.QueryContext(ctx, stmt, pq.Array(stagingFileIDs))
+	if err != nil {
+		return fmt.Errorf("cannot delete load files with staging_file_id = %+v: %w", stagingFileIDs, err)
+	}
+
+	defer func() { _ = loadLocationRows.Close() }()
+
+	if !hasUsedRudderStorage {
+		return nil // no need to delete files in rudder storage
+	}
```
Comment (achettyiitr) [inline warehouse/archive/archiver.go:422]:
```suggestion
        if !hasUsedRudderStorage {
		return nil // no need to delete files in rudder storage
	}
	stmt := fmt.Sprintf(`
		DELETE FROM %s
		WHERE staging_file_id = ANY($1) RETURNING location;`,
		pq.QuoteIdentifier(warehouseutils.WarehouseLoadFilesTable),
	)
	loadLocationRows, err := txn.QueryContext(ctx, stmt, pq.Array(stagingFileIDs))
	if err != nil {
		return fmt.Errorf("cannot delete load files with staging_file_id = %+v: %w", stagingFileIDs, err)
	}

	defer func() { _ = loadLocationRows.Close() }()
```

**Review by achettyiitr — APPROVED** [review]

**Comment by fracasula** [comment]
@lvrach I'll merge this one and address the structured logging in this file in another PR with low prio.

**Comment by fracasula** [comment]
@lvrach @achettyiitr I had a veeery quick stab and the structured logging issue [here](https://github.com/rudderlabs/rudder-server/pull/3731). It's just an idea, please have a look. If it's overkilling we can stick with the usual approach.

**Pull request merged**